torch/_inductor/pattern_matcher.py

```diff
@@ -1083,6 +1142,11 @@ def run_node(self, node: torch.fx.Node) -> Any:
                         old_node=node,
                         pass_name="Interpreter_Replacer",
                     )
+                    # This function copy-pastes the replacement graph into
+                    # the graph. If the replacement graph had any arg_kwarg_vals,
+                    # or val/tensor_meta, we propagate those over.
+                    if "arg_kwarg_vals" in node.meta:
```

Comment:
I"m a bit confused - isn't this what the above `should_propagate_arg_kwarg_vals` case is handling ?

Reply:
`replace_with_graph` has some logic where we take the replacement graph and copy it into the original graph. This step would otherwise lose the arg_kwarg_vals metadata.